Move login form to a new page

rdwebdesign opened the pull request:
**Please make sure you**
 1. Base your code and PRs against the repositories developmental branch.
 2. [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits as we enforce the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions

---
- **What does this PR aim to accomplish?:**

This PR creates a new login page, with its own layout.

![newloginpage](https://user-images.githubusercontent.com/1385443/183133712-2b6630f8-4def-4f4e-b48a-551840984b48.gif)

- **How does this PR accomplish the above?:**

Removing the old login page and creating a new one.

- **What documentation changes (if any) are needed to support this PR?:**

Maybe a few screenshots.

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

dependabot[bot] committed:
Bump actions-ecosystem/action-add-labels from 1.1.0 to 1.1.3

Bumps [actions-ecosystem/action-add-labels](https://github.com/actions-ecosystem/action-add-labels) from 1.1.0 to 1.1.3.
- [Release notes](https://github.com/actions-ecosystem/action-add-labels/releases)
- [Commits](https://github.com/actions-ecosystem/action-add-labels/compare/v1.1.0...v1.1.3)

---
updated-dependencies:
- dependency-name: actions-ecosystem/action-add-labels
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

DL6ER committed:
Add PHP-CS-Fixer Workflow

Signed-off-by: DL6ER <dl6er@dl6er.de>

DL6ER committed:
Run PHP-CS-Fixer on all files

Signed-off-by: DL6ER <dl6er@dl6er.de>

DL6ER committed:
Fix remaining styling change

Signed-off-by: DL6ER <dl6er@dl6er.de>

DL6ER committed:
Update .php-cs-fixer.dist.php

Co-authored-by: yubiuser <ckoenig@posteo.de>
Signed-off-by: DL6ER <DL6ER@users.noreply.github.com>

DL6ER committed:
Update .github/workflows/php-cs-fixer.yml

Co-authored-by: yubiuser <ckoenig@posteo.de>
Signed-off-by: DL6ER <DL6ER@users.noreply.github.com>

DL6ER committed:
Explicitly set blank_line_before_statement to exclude include* and require*

Signed-off-by: DL6ER <dl6er@dl6er.de>

DL6ER committed:
Switch to style Symfony

Signed-off-by: DL6ER <dl6er@dl6er.de>

DL6ER committed:
Fix unecessary string concat added during the conversion from PhpCsFixer to Symfony

Signed-off-by: DL6ER <dl6er@dl6er.de>

DL6ER committed:
Add setLineEnding(PHP_EOL)

Signed-off-by: DL6ER <dl6er@dl6er.de>

rdwebdesign committed:
Move login form to a new page

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser reviewed (commented):
I assume `scripts/php/loginpage.php`can go as well?

rdwebdesign committed:
Rename `header.php` to `header_authenticated.php`

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Rename `common_header.php` to `header.php`

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser commented on login.php:
```diff
@@ -0,0 +1,98 @@
+
+        <div class="card">
+            <div class="card-body login-card-body">
+                <p class="login-box-msg">Sign in to start your session</p>
```
Do we need this text at all? If so, maybe 
```suggestion
                <p class="login-box-msg">Sign in to access Pi-hole's dashboard</p>
```

rdwebdesign replied:
I don't have a strong opinion about this, but I would follow one of this ideas:
1. remove the text;
2. use "Sign in to access the dashboard"
   I don't think we need to use "Pi-hole" again. The name and logo are is already above this line.

yubiuser replied:
Then I would remove the text. What else should happen after you clicked "log in" on a login page other than being logged in?

rdwebdesign replied:
I agree.

yubiuser commented on scripts/pi-hole/php/header.php:
```diff
@@ -1,128 +1,9 @@
-$memory_usage = getMemUsage();
-
 // Retrieve layout setting from setupVars
 if (isset($setupVars['WEBUIBOXEDLAYOUT']) && !('boxed' === $setupVars['WEBUIBOXEDLAYOUT'])) {
```
I think the whole `WEBUIBOXEDLAYOUT` can go to `header_authenticated.php`

rdwebdesign replied:
I agree.
This was used in my initial attempt to create a login page, but the code changed a lot after that.

yubiuser commented on scripts/pi-hole/php/header.php:
```diff
@@ -185,152 +74,43 @@ function getTemperature()
+
+    <!-- Common styles -->
     <link rel="stylesheet" href="style/vendor/bootstrap/css/bootstrap.min.css?v=<?php echo $cacheVer; ?>">
+<?php if ($auth) { ?>
```
Do you want to leave this here or move it to `header_authenticated.php`? It won't be used on the login page.

rdwebdesign replied:
Yes.
This file (`header.php`) is included in both cases.

This `if` is needed to only use this block when authenticated.

yubiuser replied:
Yeah this is exactly my point. `header.php` is used for `login.php` and `header_authenthicated.php`. But this part is never used on the `login.php` because `$auth` is not set when the `login` is shown. Therefore it makes sense to me to move it to `header_authenticated` and remove the `if` clause.

rdwebdesign replied:
No.

`header_authenticated.php` is used ONLY for authenticated pages.
`header.php` is used in both cases (it is [included in `header_authenticated.php`](https://github.com/pi-hole/AdminLTE/blob/ac15ecaf90e2f4ca8b9228dd78f259b543c9ec15/scripts/pi-hole/php/header_authenticated.php#L138)).

rdwebdesign replied:
I'm trying to put all `<head>` stuff in one place to make it easy to maintain.
I'm also trying to group all `<scripts>` and all `<link>`  tags together in just one place.

We can remove the `if`s and separate these tags in 2 files, but I think its more logical to put them together.

yubiuser commented on scripts/pi-hole/php/header.php:
```diff
@@ -185,152 +74,43 @@ function getTemperature()
     <link rel="stylesheet" href="style/vendor/AdminLTE.min.css?v=<?php echo $cacheVer; ?>">
     <link rel="stylesheet" href="style/vendor/select2.min.css?v=<?php echo $cacheVer; ?>">
 
 <?php if (in_array($scriptname, array('groups.php', 'groups-adlists.php', 'groups-clients.php', 'groups-domains.php'))) { ?>
```
Do you want to leave this here or move it to `header_authenticated.php`? It won't be used on the login page.

rdwebdesign replied:
Same as above.

yubiuser commented on scripts/pi-hole/php/header.php:
```diff
@@ -185,152 +74,43 @@ function getTemperature()
     <script src="style/vendor/bootstrap/js/bootstrap.min.js?v=<?php echo $cacheVer; ?>"></script>
     <script src="scripts/vendor/adminlte.min.js?v=<?php echo $cacheVer; ?>"></script>
     <script src="scripts/vendor/bootstrap-notify.min.js?v=<?php echo $cacheVer; ?>"></script>
+<?php if ($auth) { ?>
```
Do you want to leave this here or move it to `header_authenticated.php`? It won't be used on the login page.

rdwebdesign replied:
Same as above.

rdwebdesign committed:
Fixes for the new login

- Remove unnecessary "Sign in to start" text;
- Adjust LCARS colors: login error message;
- Fix javascript error in `footer.js`.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser commented on login.php:
```diff
@@ -0,0 +1,98 @@
+        <strong><a href="https://pi-hole.net/donate/" rel="noopener" target="_blank"><i class="fa fa-heart text-red"></i> Donate</a></strong> if you found this useful.
+    </div>
+</div>
+<script src="scripts/pi-hole/js/footer.js?v=<?php echo $cacheVer; ?>"></script>
```
Do we really need the footer script on the login page?

![Bildschirmfoto zu 2022-08-12 09-02-57](https://user-images.githubusercontent.com/26622301/184302392-43e16f4a-badd-44d2-a8a0-8992cc9ffdf7.png)

I removed it and login still works..

rdwebdesign replied:
The only "needed" function is the one for the checkbox style (icheck).

![image](https://user-images.githubusercontent.com/1385443/184409268-bf0f34c2-5647-4336-bb07-0bdb97cdc30e.png)

yubiuser replied:
Ah ok, then we should keep it.

The error is about: 
https://github.com/pi-hole/AdminLTE/blob/3daca13bd64db0377e7399634ddc5264626a9728/scripts/pi-hole/js/footer.js#L260

rdwebdesign replied:
Because `sessionTimerCounter` is null.

I will add a test to avoid this error.

yubiuser replied:
Now it complains here: https://github.com/pi-hole/AdminLTE/blob/1714b082c3a233ea3a0c344b50352c59818676fc/scripts/pi-hole/js/footer.js#L288

> Cannot read properties of null (reading 'style')

rdwebdesign replied:
Fixed.

DL6ER committed:
Disable Yoda style

Signed-off-by: DL6ER <dl6er@dl6er.de>

rdwebdesign commented:
Superseded by #2294
